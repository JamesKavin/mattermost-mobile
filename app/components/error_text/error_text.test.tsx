// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {render} from '@testing-library/react-native';
import React from 'react';

import {Preferences} from '@constants';

import ErrorTextComponent from './index';

describe('ErrorText', () => {
    const baseProps = {
        testID: 'error.text',
        textStyle: {
            fontSize: 14,
            marginHorizontal: 15,
        },
        theme: Preferences.THEMES.denim,
        error: 'Username must begin with a letter and contain between 3 and 22 characters including numbers, lowercase letters, and the symbols',
    };

    test('should match snapshot', () => {
        const wrapper = render(
            <ErrorTextComponent {...baseProps}/>,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
