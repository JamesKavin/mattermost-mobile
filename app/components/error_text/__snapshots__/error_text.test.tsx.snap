// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText should match snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#d24b4e",
        "fontSize": 12,
        "marginBottom": 15,
        "marginTop": 15,
        "textAlign": "left",
      },
      {
        "fontSize": 14,
        "marginHorizontal": 15,
      },
    ]
  }
  testID="error.text"
>
  Username must begin with a letter and contain between 3 and 22 characters including numbers, lowercase letters, and the symbols
</Text>
`;
